### PR TITLE
Add Amazon S3 Storage Adaptor for Image Variations Event Listener

### PR DESCRIPTION
**Pull request by aliasaria**
Copying heavily from the other Image Variations storage adaptors as well as the existing S3 base storage adaptor, this is an implementation of the S3 storage adaptor for the Image Variations Event Listener

### CODE REVIEW

**Comment by rexxars** [comment]
Thanks for this! Don't suppose you feel like writing some basic tests for this?

**Comment by rexxars** [comment]
I'm not using S3 personally and don't feel comfortable merging this without any tests of any kind... Would anyone be willing to adopt this? :)

**Comment by matslindh** [comment]
We've been considering merging the "storage" requirements from the different modules into one interface - should we try to implement that as a part of this as well? I think we could do something as simple as having a namespace for the storage operations, so that different modules gets the files that are defined inside their own namespace, and the adapters can support configurations pr. namespace (i.e. 'files' => '/foo', 'default' => '/default', 'imageVariations' => '/imagevars', etc). 

If that sounds like a plan, a separate issue to track the needed changed would be a good plan.

**Comment by rexxars** [comment]
Yeah, definitely seems like a good idea, just unsure if we have the manpower to pull that of right now. I'm keen on getting 2.0 released first, and maybe schedule storage/database refactoring for 3.0?

**Comment by matslindh** [comment]
Sounds good to me. I won't have the necessary time before april/may to get through larger refactoring tasks.

**Comment by aliasaria** [comment]
Sorry guys I don't know how to write official tests :( I hope someone can take this on.

**Comment by christeredvartsen** [comment]
I can add necessary tests for this.

**Comment by christeredvartsen** [comment]
@aliasaria Can you please rebase against imbo:develop and update?

File: library/Imbo/EventListener/ImageVariations/Storage/S3.php [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        $variationsPath = $this->getImagePath($user, $imageIdentifier);
+
+        $varations = $this->getClient()->getIterator('ListObjects', [
+          'Bucket' => $this->params['bucket'],
+          'Prefix' => $variationsPath
+        ]);
+
+        // Note: could also use AWS's deleteMatchingObjects instead of
+        // deleting items one by ne
+
```
Comment (christeredvartsen) [inline library/Imbo/EventListener/ImageVariations/Storage/S3.php]: Could you implement it using `deleteMatchingObjects` instead?

**Pull request merged**